skrf/calibration/calibration.py

```diff
@@ -3470,7 +3475,424 @@ def load_calibration_archive(cls, filename):
 
         return cal
 
+class TUGMultilineTRL(EightTerm):
+    """
+    TUG Multiline TRL calibration.
+
+    An improved multiline TRL calibration procedure that generalizes the calibration process 
+    by solving a single 4x4 weighted eigenvalue problem.
+
+    The overall algorithm is based on [1]_, but the weighting matrix calculation is based on [2]_. 
+    You can read the mathematical details online at [3]_.
+
+    The calibration reference plane is at the edges of the first line. 
+    By default, the reference impedance of the calibration is the characteristic impedance 
+    of the transmission lines. If the characteristic impedance is known, 
+    the reference impedance can be renormalized afterwards by running the method `renormalize()`.
+
+    Examples
+    --------
+
+    >>> line1 = rf.Network('line1.s2p')
+    >>> line2 = rf.Network('line2.s2p')
+    >>> line3 = rf.Network('line3.s2p')
+    >>> short = rf.Network('short.s2p')
+    >>> dut   = rf.Network('dut.s2p')
+
+    Normal multiline TRL calibration:
+
+    >>> cal = rf.TUGMultilineTRL(line_meas=[line1,line2,line3], line_lengths=[0, 1e-3, 5e-3], ereff_est=4-.0j, 
+    >>>        reflect_meas=short, reflect_est=-1, reflect_offset=0)
+    >>> dut_cal = cal.apply_cal(dut)
+
+    Case of not using reflect measurements:
+
+    >>> cal = rf.TUGMultilineTRL(line_meas=[line1,line2,line3], line_lengths=[0, 1e-3, 5e-3], ereff_est=4-.0j)
+    >>> dut_cal = cal.apply_cal(dut)  # only S21 and S12 are correct
+
+    References
+    ----------
+    .. [1] Z. Hatab, M. Gadringer and W. Bösch, "Improving The Reliability of The Multiline TRL Calibration Algorithm," 
+        _2022 98th ARFTG Microwave Measurement Conference (ARFTG)_, Las Vegas, NV, USA, 2022, pp. 1-5, 
+        doi: https://doi.org/10.1109/ARFTG52954.2022.9844064
+
+    .. [2] Z. Hatab, M. Gadringer and W. Bösch, "Propagation of Linear Uncertainties through Multiline Thru-Reflect-Line Calibration," 
+            2023, e-print: https://arxiv.org/abs/2301.09126
+            
+    .. [3] https://ziadhatab.github.io/posts/multiline-trl-calibration/
+
+    See Also
+    --------
+    NISTMultilineTRL
+    """
+
+    family = 'TRL'
+    def __init__(self, line_meas, line_lengths, ereff_est=1-.0j, 
```

Comment:
The interface is different from `NISTMultilineTRL` and I think it's fine. However I would like to change `ereff_est` to `er_est` to be the same as `NISTMultilineTRL` so that they match instead of being slightly different to avoid mistakes when changing between the methods.